lib/src/hashconnect.ts

```diff
@@ -23,69 +22,73 @@ export class HashConnect implements IHashConnect {
 
     debug: boolean = false;
 
     constructor(debug?: boolean) {
-        this.relay = new WakuRelay(this);
-        
+        this.relay = new WebSocketRelay(this);
+
         this.foundExtensionEvent = new Event<HashConnectTypes.WalletMetadata>();
         this.pairingEvent = new Event<MessageTypes.ApprovePairing>();
         this.transactionEvent = new Event<MessageTypes.Transaction>();
         this.transactionResponseEvent = new Event<MessageTypes.TransactionResponse>();
         this.acknowledgeMessageEvent = new Event<MessageTypes.Acknowledge>();
         this.additionalAccountRequestEvent = new Event<MessageTypes.AdditionalAccountRequest>();
         this.additionalAccountResponseEvent = new Event<MessageTypes.AdditionalAccountResponse>();
-        
+
         this.messages = new MessageUtil();
         this.messageParser = new MessageHandler();
 
-        if(debug) this.debug = debug;
+        if (debug) this.debug = debug;
 
         this.setupEvents();
     }
 
+
     async init(metadata: HashConnectTypes.AppMetadata | HashConnectTypes.WalletMetadata, privKey?: string): Promise<HashConnectTypes.InitilizationData> {
-        this.metadata = metadata;
 
-        if(this.debug) console.log("hashconnect - Initializing")
+        return new Promise(async (resolve) => {
+            this.metadata = metadata;
 
-        if(!privKey)
-            this.privateKey = this.generateEncryptionKeys();
-        else
-            this.privateKey = privKey;
-        
-        metadata.publicKey = Buffer.from(getPublicKey(Buffer.from(this.privateKey, 'base64'))).toString('base64');
+            if (this.debug) console.log("hashconnect - Initializing")
 
-        let initData: HashConnectTypes.InitilizationData = {
-            privKey: this.privateKey
-        }
+            if (!privKey)
+                this.privateKey = await this.generateEncryptionKeys();
+            else
+                this.privateKey = privKey;
 
-        if(window)
-            this.metadata.url = window.location.origin;
+            metadata.publicKey = this.privateKey;
 
-        await this.relay.init();
+            let initData: HashConnectTypes.InitilizationData = {
+                privKey: this.privateKey
+            }
 
-        this.relay.addDecryptionKey(this.privateKey);
-        
-        return initData;
+            if (window)
+                this.metadata.url = window.location.origin;
+
+            await this.relay.init();
+
+            if (this.debug) console.log("hashconnect - Initialized")
+
+            resolve(initData);
+        });
     }
 
 
     async connect(topic?: string, metadataToConnect?: HashConnectTypes.AppMetadata | HashConnectTypes.WalletMetadata): Promise<HashConnectTypes.ConnectionState> {
-        if(!topic) {
-            if(this.debug) console.log("hashconnect - Creating new topic id");
+        if (!topic) {
             topic = this.messages.createRandomTopicId();
+            this.publicKeys[topic] = this.privateKey;
```

Comment:
Again, it might be intentional but this doesn't sound right assigning a private key to a public key variable.